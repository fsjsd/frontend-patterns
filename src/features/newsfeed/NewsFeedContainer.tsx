import React, { Suspense } from 'react'
import ContentHeader, { ContentHeaderLabel } from '../../ux/ContentHeader'
import { ContentWrapper } from '../../ux/ContentWrapper'
import Loading from '../../ux/Loading'
const NewsFeed = React.lazy(() => import('./NewsFeed'))

const NewsFeedContainer = () => {
  return (<>
    <ContentHeader>
      <ContentHeaderLabel>News feed</ContentHeaderLabel>
    </ContentHeader>
    <ContentWrapper>
      <Suspense fallback={<Loading />}>
        <div role="main">
          <NewsFeed />
        </div>
      </Suspense>
    </ContentWrapper>
  </>
  )
}

export default NewsFeedContainer