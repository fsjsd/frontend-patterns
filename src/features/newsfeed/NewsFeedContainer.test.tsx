import React from 'react'
import NewsFeedContainer from "./NewsFeedContainer";
import { render, waitFor } from '@testing-library/react'

describe("NewsFeed", () => {
  it("renders and matches snapshot", async () => {
    const { container, queryByRole } = render(<NewsFeedContainer />)
    await waitFor(() => expect(queryByRole("main")).toBeInTheDocument());
    expect(container).toMatchSnapshot();

  })
})