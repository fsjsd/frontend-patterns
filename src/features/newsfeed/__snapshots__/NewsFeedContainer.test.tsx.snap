// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsFeed renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ dtMXGe"
  >
    <div
      class="sc-gsnTZi dkPlsI"
    >
      News feed
    </div>
  </div>
  <div
    class="sc-dkzDqf eGEKCX"
  >
    <div
      role="main"
      style=""
    >
      <div
        class="sc-eCYdqJ jGnNot"
      >
        Coming soon
      </div>
    </div>
  </div>
</div>
`;
